build.gradle

```diff
@@ -0,0 +1,331 @@
+}
+
+// by default, the files go into build/docs/javadoc
+tasks.javadoc {
+	dependsOn unzipJar
+	classpath = configurations.allDeps
+	failOnError = false
+	options {
+		author = false
+		noDeprecated = false
+		noDeprecatedList = false
+		noIndex = false
+		noNavBar = false
+		noTree = false
+		splitIndex = true
+		use = true
+		version = true
+		links = [
+			"http://download.oracle.com/javase/11/docs/api/",
+			"https://www.javadoc.io/doc/com.fasterxml.jackson.core/jackson-core/2.9.9/",
+			"https://www.javadoc.io/doc/com.fasterxml.jackson.core/jackson-databind/2.9.9/"
+		]
+	}
+	source = fileTree(dir: "$src/us/kbase",
+	include: [
+		"workspace/*.java",
+		"common/service/*.java"
+	],
+	exclude: [
+		"workspace/WorkspaceServer.java",
+		"common/service/JsonServer*",
+		"common/service/JacksonTupleModule.java",
+		"common/service/JsonClientCaller.java",
+		"common/service/JsonTreeTraversingParser.java",
+		"common/service/KBaseJsonParser.java"
+	])
+	source += fileTree(dir: "$tempunpack/us/kbase/common/service", include: [
+		"UObject.java",
+		"JsonTokenStream.java",
+		"*Exception.java"
+	])
+
+	doLast {
+		delete "$tempunpack"
+	}
+}
+
+tasks.war {
+	dependsOn javadoc
+	description = "Build the WAR file. Assumes compilation has been run."
+	archiveFileName = warFile
+	destinationDirectory = file("$dist")
+
+	// location of the web.xml file
+	webXml = file("$warXml")
+
+	duplicatesStrategy "exclude"
+
+	// add the dependencies to lib
+	into("WEB-INF/lib") {
+		from sourceSets.main.compileClasspath
+	}
+	// copy the compiled classes over
+	into("WEB-INF/classes") {
+		from sourceSets.main.output.classesDirs
+	}
+	// copy in the documentation
+	into("WEB-INF/classes/server_docs/javadoc") {
+		from javadoc.destinationDir
+	}
+}
+
+task expandWar(type: Copy) {
+	from zipTree("$dist/$warFile")
+	into "war_contents"
+}
+
+// test reports are saved to build/reports/tests/<testName>
+tasks.withType(Test) {
+	dependsOn compileTestJava
+	useJUnit()
+	systemProperty "test.cfg", testConfig
+	maxHeapSize = "3G"
+	testLogging {
+		events "passed", "skipped", "failed"
+		exceptionFormat "short"
+		showStandardStreams = true
+		showExceptions = true
+		showCauses = true
+	}
+}
+
+task testNoLongTests(type: Test) {
+	description = "run all tests except the slow tests"
+	systemProperty "test.cfg", testConfig
+	forkEvery 1
+	filter {
+		excludeTestsMatching "*LongTest"
+		// gradle thinks that classes annotated with @RunWith
+		// are tests
+		excludeTestsMatching "*Tester"
+	}
+}
+
+// output from coverage tasks is saved to build/reports/jacoco/<testName>
+tasks.withType(JacocoReport) {
+	reports {
+		html.required = true
+		xml.required = true
+		csv.required = true
+	}
+	sourceSets sourceSets.main
+}
+
+task coverageNoLongTests(type: JacocoReport, dependsOn: testNoLongTests) {
+	group = "Reporting"
+	description = "Generate code coverage reports for all tests except the slow tests"
+	executionData(testNoLongTests)
+}
+
+task createScript {
```

Comment:
## Create the CLI script

Run the gradle command `gradle createScript` to generate the CLI script.

### to test:
run `ant clean && ant script`
rename the resulting file (`update_workspace_database_schema`)
run `gradle clean createScript`

replace `:` with `\n` and sort lines in both files:
```sh
sed 's/:/\n/g' update_workspace_database_schema_ant | sort > script_ant.txt
sed 's/:/\n/g' update_workspace_database_schema | sort > script_gradle.txt
```
diff the two files
```sh
diff script_ant.txt script_gradle.txt
1d0
<     
```